engine/evarutil.mli

```diff
@@ -100,6 +100,10 @@ val is_ground_env  :  evar_map -> env -> bool
     its (partial) definition. *)
 val gather_dependent_evars : evar_map -> Evar.t list -> (Evar.Set.t option) Evar.Map.t
 
+(** Like [gather_dependent_evars] but gets the [seeds] from the evars
+   appearing in the term list (including defined evars) *)
+val gather_dependent_evars_terms : evar_map -> EConstr.t list -> (Evar.Set.t option) Evar.Map.t
```

Comment:
I am not sure I am happy about exporting more broken functions in the API. The goal of the last commit was to prevent people from involuntarily relying on them, but this is just another footgun displayed on a pedestal just for one caller.